bundles/mapping/mapmodule_ol3/plugin/controls/ControlsPlugin.js

```diff
@@ -108,7 +112,36 @@ Oskari.clazz.define(
                 'DisableMapMouseMovementRequest' : mapMovementHandler
             };
         },
+        disableMouseDragZoom: function () {
+            var me = this;
+            if ( this.boxZoom ) {
+                this.getMap().removeInteraction( this.boxZoom );
+            }
+            if ( this.removedInteractions ) {
+                this.removedInteractions.forEach( function ( interaction ) {
+                    me.getMap().addInteraction( interaction );
+                });
+            }
+        },
+        mouseDragZoomInteraction: function () {
+            var me = this;
+            this.removedInteractions = [];
 
+            me.getMap().getInteractions().forEach( function( interaction ) {
+                if ( interaction instanceof ol.interaction.DragPan || interaction instanceof ol.interaction.DragZoom ) {
+                    me.getMap().removeInteraction( interaction );
+                    me.removedInteractions.push( interaction );
+                }
+            });
+
+            this.boxZoom = new ol.interaction.DragZoom( {
```

Comment:
You could wrap this with if (!this.boxZoom) since you only need to create it once.